src/content/docs/beyond-splashkit/Cryptography/1-encoding-and-decoding.mdx

```diff
@@ -0,0 +1,1302 @@
+{
+    return caesarCipherEncode(input, 26 - (shift % 26)); // Reverse the shift
+}
+```
+
+</TabItem>
+<TabItem label="Python">
+
+```python
+def caesar_cipher_decode(input_string, shift):
+    return caesar_cipher_encode(input_string, -shift)  # Reverse the shift
+```
+
+</TabItem>
+</Tabs>
+
+#### Using this function
+
+The following code demonstrates how you can use the `caesar_cipher_decode` function to decode a message:
+
+<Tabs syncKey="code-language">
+<TabItem label="C++">
+
+```cpp
+string encoded_message = "Khoor, Zruog!";
+int shift = 3;
+std::cout << "Encoded: " << encoded_message << std::endl;
+std::cout << "Decoded: " << caesar_cipher_decode(encoded_message, shift) << std::endl;
+```
+
+</TabItem>
+<TabItem label="C#">
+
+
+```csharp
+string encodedMessage = "Khoor, Zruog!";
+int shift = 3;
+Console.WriteLine("Encoded: " + encodedMessage);
+Console.WriteLine("Decoded: " + caesarCipherDecode(encodedMessage, shift));
+```
+
+</TabItem>
+<TabItem label="Python">
+
+```python
+encoded_message = "Khoor, Zruog!"
+shift = 3
+print(f"Encoded: {encoded_message}")
+print(f"Decoded: {caesar_cipher_decode(encoded_message, shift)}")
+```
+
+</TabItem>
+</Tabs>
+
+#### Expected Output
+
+The decoded message should match the original message that was encoded.
+
+```plaintext
+Encoded: "Khoor, Zruog!"
+Decoded: "Hello, World!"
+```
+
+### Brute-Forcing the Caesar Cipher
+
+In some cases, the shift value used in the caesar cipher may not be known. To decrypt the message without the key, you can use a brute-force approach to try all possible shift values and find the one that produces a readable message. As there is only 26 possible shift values in the caesar cipher, this method is feasible and can be implemented efficiently. Here is how you can brute-force the caesar cipher to decrypt a message:
+
+:::note
+This function relies on the `ceasar_cipher_decode` function to decode the message using each possible shift value, which also relies on the `caesar_cipher_encode` function.
+:::
+
+<Tabs syncKey="code-language">
+<TabItem label="C++">
+
+```cpp
+void caesar_cipher_brute_force(const string &input)
+{
+    for (int shift = 0; shift < 26; shift++)
+    {
+        std::cout << "Shift " << shift << "=> " << caesar_cipher_decode(input, shift) << std::endl;
+    }
+}
+```
+
+</TabItem>
+<TabItem label="C#">
+
+```csharp
+void caesarCipherBruteForce(string input)
+{
+    for (int shift = 0; shift < 26; shift++)
+    {
+        Console.WriteLine("Shift " + shift + "=> " + caesarCipherDecode(input, shift));
+    }
+}
+```
+
+</TabItem>
+<TabItem label="Python">
+
+```python
+def caesar_cipher_brute_force(input_string):
+    for shift in range(26):
+        print(f"Shift {shift}=> {caesar_cipher_decode(input_string, shift)}")
+```
+
+</TabItem>
+</Tabs>
+
+#### Using this function
+
+Here is an example of how you can use the `caesar_cipher_brute_force` function to decrypt a message without knowing the shift value:
+
+#### Using this function
```

Comment:
I don't believe the using this function heading should be here. In each other demonstration you only have it once whereas here its on line 612 and 616